rotkehlchen/chain/scroll/modules/scroll_bridge/decoder.py

```diff
@@ -0,0 +1,224 @@
+        Scroll -> Ethereum: Deposit to bridge
+        """
+        if (tx_log := context.tx_log).topics[0] not in (FINALIZE_DEPOSIT_ERC20, WITHDRAW_ERC20):
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(tx_log.topics[3])
+        to_address = hex_or_bytes_to_address(tx_log.data[:32])
+
+        if not self.base.any_tracked([from_address, to_address]):
+            return DEFAULT_DECODING_OUTPUT
+
+        ethereum_token_address = hex_or_bytes_to_address(tx_log.topics[1])
+        l2_token_address = hex_or_bytes_to_address(tx_log.topics[2])
+        asset = self.base.get_or_create_evm_token(l2_token_address)
+        raw_amount = hex_or_bytes_to_int(tx_log.data[32:64])
+        amount = asset_normalized_value(raw_amount, asset)
+        expected_event_type, new_event_type, from_chain, to_chain, _ = bridge_prepare_data(
+            tx_log=tx_log,
+            # The withdraw topic (Scroll to Ethereum) is considered a deposit here
+            deposit_topics=(WITHDRAW_ERC20,),
+            source_chain=ChainID.SCROLL,
+            target_chain=ChainID.ETHEREUM,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        # Find the corresponding transfer event and update it
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == to_address and
+                # Address for USDC is the gateway's address address
+                event.address in (ZERO_ADDRESS, L2_USDC_GATEWAY) and
+                event.asset == asset and
+                event.balance.amount == amount
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=from_address,
+                    to_address=to_address,
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=amount,
+                    asset=asset,
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=SCROLL_CPT_DETAILS,
+                )
+
+        log.error(
+            f'Token receiving event was not found in Scroll for '
+            f'{context.transaction.tx_hash.hex()} and L1 token {ethereum_token_address}',
+        )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_messenger_event(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a relayed message event to a bridge withdrawal"""
+        if context.transaction.input_data[:4] != RELAY_MESSAGE:
+            return DEFAULT_DECODING_OUTPUT
+
+        method_input_data = context.transaction.input_data[4:]
+
+        from_hex, to_hex, raw_amount = decode_abi(
+            ['address', 'address', 'uint256'],
+            method_input_data,
+        )
+        from_address = string_to_evm_address(to_checksum_address(from_hex))
```

Comment:
you don't need to do string_to_evm_address. Just call our deserialize function directly. `deserialize_evm_address`. That does a to_checksum, underneath.

For this and all other ones

```suggestion
        from_address = deserialize_evm_address(from_hex)
```